deploy/olm-catalog/jaeger-operator.csv.yaml

```diff
@@ -98,19 +148,15 @@ spec:
               containers:
               - args:
                 - start
-                - --platform=openshift
```

Comment:
@jpkrohling I ran the command as described [here](https://github.com/jaegertracing/jaeger-operator/blob/master/CONTRIBUTING.adoc#operator-lifecycle-manager-integration) - just wondering if the removal of the `--platform=openshift` is right?

cc @awgreene

Reply:
Doesn't look correct. OLM is, AFAIK, OpenShift-specific, so, it requires the `--platform=openshift`.

Reply:
Don't think OLM is supposed to be OpenShift specific - question is why it automatically removes this arg, but leaves the `start` argument?

Might be worth addressing this by checking for `Route` or other OpenShift specific resource to determine platform, to avoid separate deploy file?

Reply:
Hello @jpkrohling and @objectiser 

OLM and Marketplace will be able to run on vanilla Kubernetes. The Operator-sdk gen will use the `operator.yaml` by default.

Reply:
@awgreene Ok thanks.

@jpkrohling So should we just merge this PR and work on detecting the platform automatically in a separate PR?

Reply:
Yes, LGTM